fe/fe-core/src/main/java/org/apache/doris/nereids/NereidsPlanner.java

```diff
@@ -190,11 +191,14 @@ private PhysicalPlan chooseBestPlan(Group rootGroup, PhysicalProperties physical
 
         Plan plan = groupExpression.getPlan().withChildren(planChildren);
         if (!(plan instanceof PhysicalPlan)) {
-            throw new AnalysisException("generate logical plan");
+            throw new AnalysisException("Result plan must be PhysicalPlan");
         }
 
         // TODO: set (logical and physical)properties/statistics/... for physicalPlan.
-        return ((PhysicalPlan) plan).withPhysicalProperties(groupExpression.getOutputProperties(physicalProperties));
+        PhysicalPlan physicalPlan = ((PhysicalPlan) plan).withPhysicalProperties(
+                groupExpression.getOutputProperties(physicalProperties));
+        ((AbstractPlan) physicalPlan).setStats(groupExpression.getOwnerGroup().getStatistics());
```

Comment:
change withPhyscialProperties to withPhyscialPropertiesAndStats, and remove setStats function. We need to do our best to ensure immutable